examples/face_recognition_11/face_recognition_11/seed_test_suite.py

```diff
@@ -0,0 +1,125 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import sys
+from argparse import ArgumentParser
+from argparse import Namespace
+from typing import List
+from typing import Tuple
+
+import pandas as pd
+from face_recognition_11.workflow import GroundTruth
+from face_recognition_11.workflow import ImageWithMetadata
+from face_recognition_11.workflow import TestCase
+from face_recognition_11.workflow import TestSample
+from face_recognition_11.workflow import TestSuite
+
+import kolena
+
+BUCKET = "kolena-public-datasets"
+DATASET = "labeled-faces-in-the-wild"
+
+
+def create_test_case_for_tag(
+    test_samples_and_ground_truths: List[Tuple[TestCase, GroundTruth]],
+    category: str,
+    value: str,
+) -> TestCase:
+    name = f"{category} :: {value} :: {DATASET}"
+    description = f"demographic subset of {DATASET} with source data labeled as {category}={value}"
+
+    # filter down to only test samples matching this demographic
+    test_samples = [
+        (ts, gt)
+        for ts, gt in test_samples_and_ground_truths
+        if ts.a.metadata[category] == value or ts.b.metadata[category] == value
+    ]
+
+    test_case = TestCase(
+        name=name,
+        description=description,
+        test_samples=test_samples,
+        reset=True,
+    )
+
+    return test_case
+
+
+def main(args: Namespace) -> int:
+    kolena.initialize(os.environ["KOLENA_TOKEN"], verbose=True)
+
+    df = pd.read_csv(args.dataset_csv)
+    df_metadata = pd.read_csv(args.metadata_csv)
+
+    metadata_by_locator = {}
+    for record in df_metadata.itertuples(index=False):
+        fields = set(record._fields)
+        fields.remove("locator")
+        metadata_by_locator[record.locator] = {f: getattr(record, f) for f in fields}
+
+    test_samples_and_ground_truths = [
+        (
+            TestSample(
+                a=ImageWithMetadata(locator=row["locator_a"], metadata=metadata_by_locator[row["locator_a"]]),
+                b=ImageWithMetadata(locator=row["locator_b"], metadata=metadata_by_locator[row["locator_b"]]),
+            ),
+            GroundTruth(is_same=row["is_same"]),
+        )
+        for idx, row in df.iterrows()
+    ]
+
+    complete_test_case = TestCase(
+        name=f"fr 1:1 complete :: {DATASET}",
+        description=f"All images in {DATASET} dataset",
+        test_samples=test_samples_and_ground_truths,
+        reset=True,
+    )
+    print(f"created baseline test case '{complete_test_case.name}'")
+
+    # Metadata Test Cases
+    demographic_subsets = dict(
+        race=["asian", "black", "indian", "middle eastern", "latino hispanic", "white"],  # ignore "unknown"
+        gender=["man", "woman"],  # ignore "unknown"
+    )
+
+    test_cases: List[TestCase] = []
+    for category, tags in demographic_subsets.items():
+        for tag in tags:
+            test_case = create_test_case_for_tag(test_samples_and_ground_truths, category, tag)
```

Comment:
If this is slow, you could use [init_many](https://docs.kolena.io/reference/workflow/test-case/#kolena.workflow.test_case.TestCase.init_many)

Reply:
Doesn't seem slow atm